artifacts/definitions/Windows/Detection/Yara/NTFS.yaml

```diff
@@ -0,0 +1,118 @@
+name: Windows.Detection.Yara.NTFS
+author: Matt Green - @mgreen27
+description: |
+  This artifact searches the MFT, returns a list of target files then runs Yara 
+  over the target list.  
+    
+  There are 3 kinds of Yara rules that can be deployed:   
+   &nbsp;1. Url link to a yara rule.   
+   &nbsp;2. Shorthand yara in the format "wide nocase ascii:string1,string2,string3".  
+   &nbsp;3. or a Standard Yara rule attached as a parameter.  
+  Only one method of Yara will be applied and search order is as above.  
+  
+  The artifact leverages Windows.NTFS.MFT so similar regex filters can be applied 
+  including Path, Size and date. The artifact also has an option to search across 
+  all attached drives and upload any files with Yara hits.  
+    
+  Some examples of path regex may include:  
+   &nbsp;Extension at a path: Windows/System32/.+\\.dll$  
+   &nbsp;More wildcards: Windows/.+/.+\\.dll$  
+   &nbsp;Specific file: Windows/System32/kernel32\.dll$  
+   &nbsp;Multiple extentions: \.(php|aspx|resx|asmx)$  
+  Note: no drive and forward slashes.
+  
+type: CLIENT
+parameters:
+  - name: PathRegex
+    description: Only file names that match this regular expression will be scanned.
+    default: Windows/System32/kernel32\.dll$
+  - name: DriveLetter
+    description: "Target drive. Default is a C:"
+    default: "C:"
+  - name: SizeMax
+  - name: SizeMin
+  - name: AllDrives
+    type: bool
+  - name: UploadHits
+    type: bool
+  - name: EarliestSILastChanged
+    type: timestamp
+  - name: LatestSILastChanged
+    type: timestamp
+  - name: EarliestFNCreation
+    type: timestamp
+  - name: LatestFNCreation
+    type: timestamp
+  - name: YaraUrl
+    description: If configured will attempt to download Yara rules form Url
+    default:
+  - name: ShortHandYara
+    description: Second option Yara choice is a Velociraptor shorthand Yara rule
+    default: 
+  - name: YaraRule
+    description: Final Yara option and the default if no other options provided.
+    default: |
+        rule IsPE:TestRule {
+           meta:
+              author = "the internet"
+              date = "2021-03-04"
+              description = "A simple PE rule to test yara features"
+          condition:
+             uint16(0) == 0x5A4D and
+             uint32(uint32(0x3C)) == 0x00004550
+        }
+
+sources:
+  - precondition:
+      SELECT OS From info() where OS = 'windows'
+
+    query: |
+      -- check which Yara to use
+      LET yara = SELECT * FROM if(condition=YaraUrl,
+            then= { SELECT Content FROM http_client( url=YaraUrl, method='GET') },
+            else= if(condition=ShortHandYara,
+                then= { SELECT ShortHandYara as Content FROM scope() },
+                else= { SELECT YaraRule as Content FROM scope() }))
+            
+      -- first find all matching files mft
+      LET files = SELECT IsDir,
+            split(string=MFTPath,sep='\\$')[0] +  FullPath as FullPath
```

Comment:
Is this still needed? Shouldn't we just use the drive letter here?

Reply:
This is still required as if the user used the AllDrives option the only way to determine the Drive is with the split function.